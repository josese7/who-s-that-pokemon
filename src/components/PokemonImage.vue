<template>
    <div class="pokemon-container">
        <img   v-if="!showPokemon" class= "pokemon-hidden" :src="imgSrc" alt="">
        <img  v-else class="fade-in" :src="imgSrc" alt="">
        

    </div>
</template>

<script>
export default {
    props:{
        pokemonId:{
            type: Number,
            require:true

        },
        showPokemon:{
            type: Boolean,
            required: true
        }
    },
    computed:{
        imgSrc() {
            return `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${this.pokemonId}.svg`
        }
    }
}
</script>

<style scoped>
.pokemon-container {
    height: 20px;
}
img {
    height: 200px;
    position: absolute;
    left: 0;
    right: 0;
    margin: 0 auto; 
    user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    -webkit-user-drag: none;
    -webkit-user-select: none;
}
.pokemon-hidden {
    filter: brightness(0);
}
</style>