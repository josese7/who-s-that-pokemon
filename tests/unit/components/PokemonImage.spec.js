import { shallowMount } from "@vue/test-utils";
import PokemonImage from '@/components/PokemonImage'

describe('PokemonImage Component',()=>{

    test('snapshot', () => { 
        const wrapper = shallowMount(PokemonImage, {
            props:{
                showPokemon: false
            }
        })

        expect(wrapper.html()).toMatchSnapshot()
     })
    test('debe mostrat la imagen oculta del pokemon 100', () => { 
        const wrapper = shallowMount(PokemonImage, {
            props:{
                showPokemon: false,
                pokemonId:100
            }
        })

        const [img1, img2] = wrapper.findAll('img')

        expect(img1.exists()).toBeTruthy()
        expect(img2).toBe(undefined)
        console.log(img1.clases)

        expect(img1.classes('pokemon-hidden')).toBeTruthy()

        expect(img1.attributes('src')).toBe('https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg')

    })
    test('debe mostrar el pokemon, si showPokemon es true', () => { 

     })
})