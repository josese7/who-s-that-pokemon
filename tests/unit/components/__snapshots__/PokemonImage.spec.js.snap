// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonImage Component snapshot 1`] = `<div class="pokemon-container"><img class="pokemon-hidden" src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/undefined.svg" alt=""></div>`;
